identity/src/main/kotlin/net/thechance/identity/service/IdentityImageStorageService.kt

```diff
@@ -0,0 +1,62 @@
+package net.thechance.identity.service
+
+import net.thechance.identity.exception.InvalidImageException
+import net.thechance.identity.exception.UnknownErrorException
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.stereotype.Service
+import org.springframework.web.multipart.MultipartFile
+import software.amazon.awssdk.core.sync.RequestBody
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.ObjectCannedACL
+import software.amazon.awssdk.services.s3.model.PutObjectRequest
+import java.time.LocalDateTime
+
+@ConfigurationProperties(prefix = "storage.mena")
+data class IdentityStorageProperties(
+    val bucket: String,
+    val cdnEndpoint: String
+)
+
+@Service
+@EnableConfigurationProperties(IdentityStorageProperties::class)
+class IdentityImageStorageService(
+    private val menaS3Client: S3Client,
+    private val identityStorageProperties: IdentityStorageProperties,
+) {
+    fun uploadImage(
+        file: MultipartFile,
+        fileName: String,
+        folderName: String,
+    ): String {
+        val mimeType = file.contentType ?: throw InvalidImageException("null")
+        val extension = allowedMimeTypes[mimeType] ?: throw InvalidImageException(mimeType)
+        try {
+            val fileName = "${fileName}_${LocalDateTime.now()}.$extension"
+            val key = "images/$folderName/$fileName"
```

Comment:
lets add Identity folder here
```suggestion
            val key = "images/identity/$folderName/$fileName"
```

Reply:
done